frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -198,11 +219,26 @@ public long getStartTime() {
         return startTime;
     }
 
-    public void shutdown() {
+    public void shutdown(boolean falseStop) {
         running.set(false);
         backendChannel.close();
         if (currentThread != null) {
             currentThread.interrupt();
+            try {
```

Comment:
Ahh. Ok. The concern is that this would increase the batch wait time. We could do with both appraoches. But adding it back to the front of the queue might be a more user friendly option.